packages/blocks/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -0,0 +1,134 @@
+  }
+
+  const handleKeyPress = (e: KeyboardEvent) => {
+    if (e.key === 'Enter' && value.length > 0) {
+      setState([...state, { title: value, checked: true }]);
+
+      if (getState) {
+        getState([...state]);
+      }
+      setValue('');
+    }
+  }
+
+  useEffect(() => {
+    window.addEventListener('click', handleClick, false);
+    window.addEventListener('keypress', handleKeyPress, false);
+
+    return () => {
+      window.removeEventListener('click', handleClick);
+      window.removeEventListener('keypress', handleKeyPress);
+    }
+  }, []);
+
+  return (
+    <div className="t-container">
+      <div className="t-tags-container">
+        {state.map((item, index) => {
+          return (item.checked) ? (
+            <Tag
+              skin="gray"
+              key={`key_${index}`}
+              onClose={() => {
+                item.checked = !item.checked;
+                setState([...state]);
+
+                if (getState) {
+                  getState([...state]);
+                }
+              }}
+            >
+              {item.title}
+            </Tag>
+          ) : null;
+        })}
+      </div>
+      <div className={cx('t-search-container', {'t-shadow': isActive})}>
+        <label
+          className={cx('t-input-container', {'active': isActive})}
+          onClick={() => setIsActive(true)}
+        >
+          <SearchIcon className="t-icon" />
+          <input
+            type="text"
+            value={value}
+            onChange={(e) => {
+              setValue(e.target.value);
+            }}
+          />
+        </label>
+        {isActive && (
+          <div>
+            <div className="p-3">
+              {state.map((item, index) => {
+                return item.title.includes(value) ? (
+                  <Checkbox
+                    key={`key2_${index}`}
+                    label={item.title}
+                    checked={item.checked}
+                    onChange={(e) => {
+                      state[index].checked = !state[index].checked;
+                      setState([...state]);
+
+                      if (getState) {
+                        getState([...state]);
+                      }
+                    }}
+                  />
+                ) : null;
+              })}
+            </div>
+            <div className="t-add-container">
+              <button
+                onClick={() => {
+                  if (value.length > 0) {
+                    setState([...state, { title: value, checked: true }]);
+
+                    if (getState) {
+                      getState([...state, { title: value, checked: true }]);
+                    }
+
+                    setValue('');
+                  }
+                }}
+              >
+                <PlusIcon className="t-icon" />
+                <p className="ml-3">Create {value}</p>
+              </button>
+            </div>
+          </div>
+        )}
+      </div>
+    </div>
+  );
+};
```

Comment:
When I click on a part of the pop-up area that's _not_ a checklist item, it closes the menu, which seems like odd behavior. Only clicks outside the menu area should close it

<img width="1754" alt="image" src="https://user-images.githubusercontent.com/57259/163727197-4f0c1b8e-d112-442b-bb1c-81f83948b03d.png">